config/routes.rb

```diff
@@ -0,0 +1,31 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  root 'works#homepage'
+
+  resources :votes, only: [:index, :show, :create]
+
+  resources :works do
+    resources :votes, only: [:index, :create, :update]
+    member do
+      post 'upvote'
+    end
+  end
+
+  resources :works do
+
+   end
+
+
+  resources :users do
+    resources :votes, only: [:index, :create]
+  end
+
+  resources :sessions, only: [:new, :create]
+
+  post '/sessions/logout', to: 'sessions#logout', as: 'logout'
```

Comment:
Nice! You had a specific action and URL that you wanted to make as a custom route, you knew which controller and action it went to, and the name you would refer to it within the app code.